src/fu-console.c

```diff
@@ -0,0 +1,753 @@
+/*
+ * Copyright (C) 2017 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#define G_LOG_DOMAIN "FuProgressBar"
+
+#include "config.h"
+
+#include <glib/gi18n.h>
+#include <stdio.h>
+
+#include "fu-console.h"
+
+#ifdef _WIN32
+#include <wchar.h>
+#include <windows.h>
+#endif
+
+struct _FuConsole {
+	GObject parent_instance;
+	GMainContext *main_ctx;
+	FwupdStatus status;
+	gboolean spinner_count_up; /* chars */
+	guint spinner_idx;	   /* chars */
+	guint length_percentage;   /* chars */
+	guint length_status;	   /* chars */
+	guint percentage;
+	GSource *timer_source;
+	gint64 last_animated; /* monotonic */
+	GTimer *time_elapsed;
+	gdouble last_estimate;
+	gboolean interactive;
+	gboolean tainted;
```

Comment:
this is a weird adjective to indicate it's got contents to clear.  Any better ideas?

Reply:
I think you do :) `contents_to_clear` works for me, fixed.